test/sql/types/union/union_extract.test

```diff
@@ -139,3 +139,13 @@ query III
 from (values('asdf'::UNION(a INTEGER, b VARCHAR, c FLOAT))) v(c) select c.*;
 ----
 NULL	asdf	NULL
+
+statement error
+FROM (VALUES (row(1,2))) as t(i) SELECT i.*;
+----
+SELECT list is empty
```

Comment:
This is now still allowed but throws an error in this specific situation, the following query does complete which seems unintentional to me:

```sql
D FROM (VALUES (row(1,2))) as t(i) SELECT i.*, 42;
┌───────┐
│  42   │
│ int32 │
├───────┤
│  42   │
└───────┘

```

Perhaps we can explicitly throw when `i.*` is used on an unnamed struct (`StructType::IsUnnamed`)?